source/Octostache/Templates/Functions/TextManipulationFunction.cs

```diff
@@ -95,5 +96,68 @@ public static string Trim(string argument, string[] options)
                     return null;
             }
         }
+
+        public static string UriPart(string argument, string[] options)
+        {
+            if (argument == null)
+                return null;
+
+            if (!options.Any())
+                return $"[{nameof(UriPart)}: no argument given]";
```

Comment:
I actually wonder about a slight enhancement here. What if we emitted `[{nameof(UriPart) error: no argument given]`? (Adding the word `error`) This way the word error can be searched for in logs, and it's clear there is a problem.
```suggestion
                return $"[{nameof(UriPart)} error: no argument given]";
```

Reply:
I've updated the code with the suggested change.